testing/trino-testing/src/main/java/io/trino/testing/datatype/CreateAndInsertDataSetup.java

```diff
@@ -80,16 +82,22 @@ protected String tableDefinition(List<ColumnSetup> inputs)
                     .mapToObj(column -> format("col_%d %s", column, inputs.get(column).getDeclaredType().orElseThrow()))
                     .collect(joining(",\n", "(\n", ")"));
         }
+        if (inputs.stream().noneMatch(input -> input.getDeclaredType().isPresent())) {
+            // This should use smarter check of underlying system as soon as tests need to use e.g. `getQueryRunner()::execute` as an SqlExecutor.
+            checkState(
+                    sqlExecutor instanceof TrinoSqlExecutor,
+                    "Explicit declared types are required when creating a table in a system other than Trino. " +
+                            "This is to avoid ambiguities that could occur if same-looking literal has a different type " +
+                            "in various systems we connected to in tests.");
```

Comment:
I'd say this instead:
> Explicit declared input types are required when creating a table from any non-Trino system to prevent confusion between different system's type literals.